src/components/MAPI/clusters/ClusterListItemDetails.tsx

```diff
@@ -0,0 +1,63 @@
+import { Box, Text } from 'grommet';
+import { useHttpClient } from 'lib/hooks/useHttpClient';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import * as capiv1alpha3 from 'model/services/mapi/capiv1alpha3';
+import * as capzv1alpha3 from 'model/services/mapi/capzv1alpha3';
+import * as metav1 from 'model/services/mapi/metav1';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import { getLoggedInUser } from 'stores/main/selectors';
+import useSWR from 'swr';
+
+interface IClusterListItemDetailsProps {
+  cluster: capiv1alpha3.ICluster;
+}
+
+const ClusterListItemDetails: React.FC<IClusterListItemDetailsProps> = ({
+  cluster,
+}) => {
+  const client = useHttpClient();
+  const user = useSelector(getLoggedInUser);
+
+  // Use type intersection for multiple cluster types.
+  const { data, error } = useSWR<capzv1alpha3.IAzureCluster, GenericResponse>(
```

Comment:
For multiple providers we can have

```suggestion
  const { data, error } = useSWR<capzv1alpha3.IAzureCluster | capav1alpha3.IAWSCluster | ..., GenericResponse>(
```